src/Tracing/TracingServiceProvider.php

```diff
@@ -28,12 +29,12 @@ static function (): string {
 
         $eventsDispatcher->listen(
             ManipulateAST::class,
-            Tracing::class . '@handleManipulateAST'
-        );
-
-        $eventsDispatcher->listen(
-            StartRequest::class,
-            Tracing::class . '@handleStartRequest'
+            static function (ManipulateAST $manipulateAST): void {
+                ASTHelper::attachDirectiveToObjectTypeFields(
+                    $manipulateAST->documentAST,
+                    Parser::constDirective('@tracing')
```

Comment:
An optimization could be to maybe put the `Parser::constDirective('@tracing')` outside the closure so it only has to parse it once instead of for every object type in the schema. I'm not a 100% sure it matters (especially with schema caching not) but still.

Reply:
done https://github.com/nuwave/lighthouse/commit/b09492276e175fad34e33d60f172055704ec7f2b